### ISSUE TITLE
More flexible handling for boolean arguments

### ISSUE THREAD

**Issue by AhmedSoliman** [facebookincubator/python-nubia#36]
In argparse, we only generate an argument for the boolean to set to _true_ but there is no way for the user to explicitly switch it off (pass False).

Our options:
1. Generate a '--no-X' argument automatically to toggle the boolean. (opinionated decision)
2. Give the user better control through the `@argument` to define the behaviour (needs spec)

**Comment by toti1212** [comment]
Hi @AhmedSoliman 
I think other CLI frameworks take the 2nd approach, isn't it? For example, [oclif](https://oclif.io/docs/flags) permits to change the default behavior.

**Comment by AhmedSoliman** [comment]
@toti1212 Correct, this needs a design proposal though that we can agree on before we jump to implementation.
